src/abstractdatatable/abstractdatatable.jl

```diff
@@ -542,35 +535,13 @@ function Base.convert{T}(::Type{Matrix{T}}, dt::AbstractDataTable)
     res = Matrix{T}(n, p)
     idx = 1
     for (name, col) in zip(names(dt), columns(dt))
-        any(isnull, col) && error("cannot convert a DataTable containing null values to array (found for column $name)")
+        !(T >: Null) && any(isnull, col) && error("cannot convert a DataTable containing null values to array (found for column $name)")
```

Comment:
I don't think the type check is backwards? We're saying, "hey, if the `T` you're trying to convert to isn't a `Uniont{T, Null}` and you have nulls in your dataset: error!"